client/lib/directly/index.js

```diff
@@ -14,45 +20,64 @@ const DEFAULT_RTM_WIDGET_OPTIONS = {
+	window.DirectlyRTM = window.DirectlyRTM || function() {
+		( window.DirectlyRTM.cq = window.DirectlyRTM.cq || [] ).push( arguments );
+	};
+	// Since we can only configure once per pageload, this library only provides a
+	// single global configuration.
+	window.DirectlyRTM( 'config', DEFAULT_RTM_WIDGET_OPTIONS );
+}
 
+/**
+ * Inserts a dummy DOM element that Directly uses to calculate the base URL for its assets.
+ *
+ * Under standard setup, Directly gathers its base URL by inspecting the src attribute of
+ * the <script> used to load the library. Since we've got a custom setup that doesn't include
+ * a <script> tag, we need to fake this by placing a DOM element with id="directlyRTMScript"
+ * and a src with the appropriate base URL.
+ *
+ * @see https://cloudup.com/cySVQ9R_O6S
+ */
 function insertDirectlyDOM() {
-	// Directly gathers the base URL for its assets by inspecting the src attribute
-	// of a DOM element with id="directlyRTMScript". Since we're not using the standard
-	// integration, we need to fake it by adding a dummy DOM object with those attributes.
 	const d = document.createElement( 'div' );
 	d.id = 'directlyRTMScript';
 	d.src = DIRECTLY_ASSETS_BASE_URL;
 	document.body.appendChild( d );
 }
 
+/**
+ * Initializes Directly if it hasn't already been initialized, and then executes the
+ * command by passing the arguments to window.DirectlyRTM
+ *
+ * @see initialize
+ *
+ * @returns {Promise} Promise that resolves after initialization and command execution
+ */
 function executeDirectlyCommand( ...args ) {
-	// Havoc is wreaked if you try to execute Directly commands before initialization.
-	// If initialization hasn't started, ignore the command.
-	if ( typeof initializationPromise === 'undefined' ) {
-		return;
-	}
-	// If initialization has started, add the command to the promise chain to execute
-	// as soon as initialization has finished.
-	return initializationPromise.then( () => {
-		window.DirectlyRTM( ...args );
-	} );
+	return initialize().then( () => window.DirectlyRTM( ...args ) );
 }
 
-export function initialize( directlyConfig = {} ) {
-	if ( typeof initializationPromise !== 'undefined' ) {
-		return initializationPromise;
-	}
-	// Set up the global DirectlyRTM function, required for the Directly library
-	window.DirectlyRTM = window.DirectlyRTM || function() {
-		( window.DirectlyRTM.cq = window.DirectlyRTM.cq || [] ).push( arguments );
-	};
-	// Before loading the script, we need to enqueue the config details...
-	window.DirectlyRTM( 'config', { ...DEFAULT_RTM_WIDGET_OPTIONS, ...directlyConfig } );
-	// ... and prepare the DOM for Directly
-	insertDirectlyDOM();
+/**
+ * Initializes Directly if it hasn't already been initialized. This sets up global
+ * objects and DOM elements and requests the vendor script.
+ *
+ * @returns {Promise} Promise that resolves after initialization completes
+ */
+export function initialize() {
+	return _initialize = _initialize || new Promise( ( resolve, reject ) => {
```

Comment:
Hm, to me the variable name `directly` feels confusing... If I saw this line in isolation it wouldn't seem verbose enough to know what's going on:
```
directly.then( () => window.DirectlyRTM( ...args ) );
```

If a Promise is supposed to represent an eventual value, then what we're representing is a boolean for whether the library is ready to be used. So shouldn't it be more like `isReady`?

```
isReady.then( () => window.DirectlyRTM( ...args ) );
```

Am I off base here? I'm not super invested in this particular case since I think the `initialize` function is the only thing that will interact with this variable, and everything else should use:

```
initialize().then( () => window.DirectlyRTM( ...args ) );
```